docs/src/content/docs/reference/lake-formation-tbac-recommended-strategy.mdx

```diff
@@ -0,0 +1,158 @@
+have multiple tags, with a maximum of **50 tags** and **no duplicate keys**.
+- Grants are applied **TO** principals **ON** specific tags. For example:
+
+  ```sql
+  GRANT ACCESS ON TAGS foo=bar TO user
+    ```
+
+- In a grant expression, all tag **keys** are evaluated using `AND` logic, while tag **values** are
+evaluated using `OR` logic. For instance:
+
+
+  The user has access to resources tagged with both `foo=bar` AND `spam=eggs`
+  ```sql
+  GRANT ACCESS ON TAGS foo=bar AND spam=eggs TO user
+  ```
+
+  The user has access to resources tagged with either `foo=bar` OR `foo=baz`
+  ```sql
+  GRANT ACCESS ON TAGS foo=['bar', 'baz'] TO user
+  ```
+- Tags assigned to resources are **inherited**, unless specifically overridden.
+
+  ![Tag Inheritance Diagram](imgs/tag_inheritance.png)
+- Grants ensure access to resources where **ALL** conditions specified are true.
+
+  ![Grants Diagram](imgs/grants_diagram.png)
+
+
+### Limitations
+
+There are a few limitations in designing flexible tag systems:
+
+- A resource cannot have the same LF-tag key more than once. For example, you cannot apply both
+`team=sales` and `team=marketing` to the same table.
+
+  One workaround is to embed the name inside the tag key, making the tag behave as a toggle, such as
+  `team:sales=true` and `team:marketing=true`.
+
+- A `GRANT` statement cannot use an `OR` condition across different tag keys. The following statement is
+**invalid**:
+
+  ❌ Invalid
+  ```sql
+  GRANT ACCESS ON TAGS foo=bar OR spam=eggs TO user
+  ```
+
+  ✅ Valid
+  ```sql
+  GRANT ACCESS ON TAGS foo=bar TO user
+  GRANT ACCESS ON TAGS spam=eggs TO user
+  ```
+
+
+### Best Practices
+
+Here are some best practices and recommendations to consider when designing your tagging system:
+
+- LF-Tags are hierarchical. If you apply a `GRANT` statement to a high-level resource (like a database),
+this inherently grants access to all child resources (like a table) that share the same tag value.
+
+- When creating a `GRANT` statement, remember that tags are **AND-ed**, not **OR-ed**. For example:
+
+  ```sql
+  GRANT ACCESS ON TAGS team:marketing=true TO executives
+  ```
+
+  This grants the `executives` group access to all resources tagged with `team:marketing=true`.
+
+  If the statement is modified to:
+
+  ```sql
+  GRANT ACCESS ON TAGS team:marketing=true AND PII=true TO executives
+  ```
+
+  The `executives` group is granted access only to resources that are tagged with both `team:marketing=true`
+  and `PII=true`.
+
+## Tag Strategy
+
+There is no one-size-fits-all solution; the most effective strategy is the one that aligns best with an
+organization's structure. The recommended strategy is a modified version of the
+[AWS Recommended Common LF-Tag Ontologies](https://aws.github.io/aws-lakeformation-best-practices/lf-tags/common-ontologies/).
+
+Below are some suggested LF-tagging conventions:
+
+- `product:<product_name> = true`: Apply this tag to resources that belong to a specific product.
```

Comment:
But 1 team can work on many different projects/products? I think we need to include it, or something lower level that identifies what teams work on. It's still a lot less than the AWS recommended ones, which is like 9 I think

Reply:
>something lower level that identifies what teams work on

we don't need to identify what "product" something is. This is not an exercise in semantics.

If there are assets A, B, C, D and E? Tag them with what team needs to work upon them. Product tag is redundant.

A: team:data and team:marketing
B: team:data and team:sales
C: team:sales and team:design
D: team:sales and team:marketing
E: team:marketing and team:design

You can go very far with this, without having to add a `product` tag that's vague and severly overlaps with `team`.

Reply:
There is only overlap if you have 1 Team and 1 Product for ever. Which will hardly be the case. Example, the DataEngineering team can work on many projects/product lines inside the company. How can you then give access of ProjectA but not ProjectB to someone else? You won't be able to

Reply:
> How can you then give access of ProjectA but not ProjectB to someone else? You won't be able to

What? Of course you can. I don't get your point. If teamX needs to have access on several assets, just tag those assets with `teamX`.

Reply:
So your solution to: 

- Project 1 Database (team:de)
- Project 2 Database (team:de)

And if you want to give the Project 2 Database access to say the Marketing team, you then add tags to the resources:
- Project 1 Database (team:de)
- Project 2 Database (team:de, team:marketing)

But what if there are many resources? Your solution is for them to add new tags on all resources that they then require. Where as if those tags were already there,  all that is needed is Data lake permissions (grants) instead of adding the tag to potentially many many resources every time someone new wants access. That is the advantage of tags.

Reply:
If there are many resources you'd still have to tag them anyway, how else would anybody get access otherwise? I am really not getting your point.

Reply:
Ah, wait, I understand now. It's still quite overlapping with the `team` functionality and might end up in becoming a spaghetti.

Imho, one should get started with either `product` or `team`, but not together. Up to you in the end, it's your project.

Reply:
I'm busy adding an example to the code: 

We suggest starting with a basic tagging strategy and limiting the number of tags:

- `team:<team_name> = true`: Apply this tag to resources belonging to a specific team.
- `sensitive = [true, false]`: For resources containing sensitive data, such as Personally Identifiable
Information (PII).
- `sharable = [true, false]`: For resources that can be shared with other teams.

Start by tagging on a Database level and apply the above tags on each. Adding a `team` tag for every team that needs access to that resource. If a new team needs access, you can simply add a tag to that Database with that `team` name resource.

This model works if you want that team to have the same access over all the resources that it has access to. This model breaks down when:
- You have many resources that need to be shared with many teams. Adding the team's tag to each resource is time consuming and goes against what tags try to solve.
- You don't want your permission grant for that team's tag to be the same on all resources.

Let's see this in action with an example.

We have the following teams:
- `DataEngineering`
- `DataScience`

We have the following resources:
- `Database1` with tags (`sensitive=false`, `sharable=true`, `team:DataEngineering=true`)
- `Database2` with tags (`sensitive=false`, `sharable=true`, `team:DataEngineering=true`)

With the following grant for the `DataEngineering` team and the `DataScience` team:
```sql
GRANT FULL ACCESS ON TAGS team:DataEngineering=true TO RoleDateEngineering
GRANT FULL ACCESS ON TAGS team:DataScience=true TO RoleDataScience
```

The DataScience team now requires access to `Database2`. You can simply add the `team:DataScience=true` tag to`Database2` and the DataScience team will have access to it. So now the tags look like:
- `Database1` with tags (`sensitive=false`, `sharable=true`, `team:DataEngineering=true`)
- `Database2` with tags (`sensitive=false`, `sharable=true`, `team:DataEngineering=true`,team:DataScience=true`)

But this has two problems, you have to apply tags instead of grants every time a new team needs access to a resource and that team will then have the same access to all resources that it's tag provides access to.

Reply:
I was starting with the "so then we recommend adding the `product` tag to solve the problem. But if we recommend that the product tag is used instead of the team tag then it solves the whole problem from the beginning. I'll push when I'm done then you can see

Reply:
>then it solves the whole problem from the beginning

Yeah indeed

Reply:
Actually wait. It changes depending on how one looks at it:

- product tag: possibly many and you don't know them all in advance. You need to re-run DLZ to give permissions.
- team tag: likely few, you know them all in advance but you don't know where they will be applied. You don't need to re-run DLZ to give permiossions, you just need to tag related resources.

Both approaches have a tradeoff.

Reply:
Well, there goes my day... 

I have rewritten `Tag strategy` https://github.com/DataChefHQ/aws-data-landing-zone/pull/95/files#diff-476e5c2d56d8a9123dfa46460efd70f859583334be0337b272e51df72266a5b1R115 to reflect what we have been talking

Reply:
oh damn man ok, didn't intend for you to rewrite all 😅. But good job, thanks for this.

Reply:
Can we merge it?

Reply:
yeye of course